### DISCUSSION TITLE
Volatile for Captured Variables

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#6431] [Language Ideas] [answered]
When using lambdas, captured variables are promoted to fields. 

Please allow marking variables as "volatile" so that promoted variables gain this.

**Comment by bernd5** [comment]
Why you don't use a custom struct / class with the fields you need?
Currently we can't specify captured variables...

**Comment by HaloFour** [comment]
IMO the C# spec doesn't state how a lambda is implemented, including captured locals, but this proposal would seemingly want to influence those implementation details in a manner that I think would cause those details to necessarily leak into the spec.  Maybe the definition of `volatile` in this case could be interpreted somewhat loosely to avoid that.

**Comment by Joe4evr** [comment, marked as answer]
You can use `Volitile.Read`/`.Write` on the local you're capturing to get the same semantics today.
[SharpLab](https://sharplab.io/#v2:C4LgbgpgTglgZgTwD4AEBMBGAsAKBQBgAIUMA6AFQAsoIBDAExgDsBzAbl1xQGZi1CAIhDjQa9XAG9chGYQAOsMLWARCNBgHsmAGwTEM/APq0AxsBhaOOWcV5CRUMQAoS/U+a0BKQgF4AfITGZhZMvoTuIVY2PMQALIQAkkxgGgDWEE7e/oERWplWAL6ceBgAbHyEAAqS0rIk5SjxALK0zJm1MlLWNrLMwITaGia02mGxaFE9MkpQhPTCYUwQAO6CwqIQ9E6ZvgEkAJxOAGoa2sow2hCkAEp0WzRwA0Mjnq+TU4PDoz6EaAAc7x68zgpCSKXS+Q6hAKNlwRRwQA=)

**Reply by TonyValenti** [reply]
Thanks @Joe4evr .  This is exactly what I needed.
